zookeeper-server/src/main/java/org/apache/zookeeper/cli/DelQuotaCommand.java

```diff
@@ -109,17 +123,11 @@ public static boolean delQuota(ZooKeeper zk, String path,
         } catch (IllegalArgumentException ex) {
             throw new MalformedPathException(ex.getMessage());
         } catch (KeeperException.NoNodeException ne) {
-            System.err.println("quota does not exist for " + path);
-            return true;
+            throw new KeeperException.NoNodeException(ne.getMessage());
         }
         StatsTrack strack = new StatsTrack(new String(data));
```

Comment:
This assumes that [the "platform's default charset"](https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#String-byte:A-) is ASCII-compatible.  I would suggest being explicit about [US-ASCII](https://docs.oracle.com/javase/8/docs/api/java/nio/charset/StandardCharsets.html#US_ASCII) here:

```java
import static java.nio.charset.StandardCharsets.US_ASCII;
[...]
new String(data, US_ASCII)
```

Also: there are quite a few instances of `new StatsTrack(new String(data))` in this patch; wouldn't it be better to just add a constructor which takes a `byte[]`?

Reply:
yes and maybe UTF-8 is better

Reply:
Done 
1. set UTF-8 
2. add new constructor `public StatsTrack(byte[] stat) `

Reply:
Actually, the content of quota znode is generated by server, it only contains character, figure. so no garbled codes?